### PR TITLE
Specs integration

### PR DESCRIPTION
**Pull request by EddyShimwa**
## Hello there! In this PR we were able to:

- create `user_index_spec.rb`
- create `user_show_spec.rb`
- create `post_index_spec.rb`
- create `post_show_spec.rb`
- secure `app from n+1 problems.`

in order to test the spec files on terminal type the following:
`rspec spec/views --format documentation`

### before n+1

Screenshot 2023-04-19 at 1 03 26 PM

### after n+1 

Screenshot 2023-04-19 at 1 06 02 PM

Thank you for reviewing our code 😊

### CODE REVIEW

File: spec/views/post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    expect(page).to have_content('3')
+  end
+
+  it 'Show the number of likes on a post' do
+    expect(page).to have_content('0')
+  end
+
+  it 'Show the button Pagination' do
+    expect(page).to have_content('Pagination')
+  end
```
Comment (Qoosim) [inline spec/views/post_index_spec.rb:79]: - [ ] Kindly fix the issue with this test. It's failing at the moment.

show-pagination

File: spec/views/post_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    expect(page).to have_content('3')
+  end
+
+  it 'Show the number of likes' do
+    expect(page).to have_content('1')
+  end
+
+  it 'Show the post body' do
+    expect(page).to have_content('MyText1')
+  end
```
Comment (Qoosim) [inline spec/views/post_show_spec.rb]: - [ ] Kindly make this test pass. It's failing at the moment

post-body

File: spec/views/user_show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+  it "When I click on a user post, it redirects me to that post's show page" do
+    click_link 'See all posts'
+    expect(page).to have_current_path user_posts_path(@user_1)
+  end
+  it "When I click to see all posts, it redirects me to the user's post's index page" do
+    click_link 'Title1'
+    expect(page).to have_current_path user_post_path(@user_1, @post_1)
+  end
+end
```
Comment (Qoosim) [inline spec/views/user_show_spec.rb]: - [ ] These set of tests are failing in its entirety. Kinldy look into them and fix.

- Failure #1 
failure-1


- Failure #2 

failure-2


- Failure #3

failure-3

- Failure #4 
failure-4

-  Failure #5 
failure-5

-  Failure #6 
failure-6

-  Failure #7 
failure-7

-  Failure #8 
failure-8

-  Failure #9
failure-9

-  Failure #10
failure-10

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Git Flow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by youmari — APPROVED** [review]
# Project Approved :trophy: :tada: 🟢 

Hi @EddyShimwa 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linters error.
:heavy_check_mark: Details PR title and description.
:heavy_check_mark: Good commit messages.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U02FTTCCU2V)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by EddyShimwa** [comment]
Thanks @youmari and @Qoosim for reviewing my PR

**Pull request merged**